scipy/stats/_distn_infrastructure.py

```diff
@@ -3362,6 +3363,8 @@ def __init__(self, a=0, b=inf, name=None, badvalue=None,
 
         if len(xk) != len(pk):
             raise ValueError("xk and pk need to have the same length.")
+        if not np.allclose(np.size(np.where((np.array(pk) >= 0.0) & (np.array(pk) <= 1.0))), np.size(pk)):
```

Comment:
This line is too long for PEP8, also you shouldn't compare integers with allclose. You can simply use `!=` and get rid of np.sizes
```python
if np.where((np.array(pk) >= 0.0) & (np.array(pk) <= 1.0))).size != len(pk):
    ...
```

Reply:
Ah, yes, it then reduces even further to

```
if all([x >= 0. for x in pk]):
    ....
```

Reply:
Oh, that would be nice. 
I though about an algorithm too.

BUT this `if condition` is aim to raise a error for range of pk.
To simplify the code, I agree with all. 
In that case, if pk is negative, error is happen.
If pk is greater than 1.0, however, error is not happen (I know that `np.sum(pk)` can catch this error).

How do you think about?

Reply:
Elements of `pk` cannot be negative if we test for `>= 0.` . So *first all elements are negative* condition is established. If elements are greater than `>= 1.` then their sum cannot be `<= 1.`

Reply:
ok, I will change error message to

```python
if not all([x >= 0.0 for x in pk]):
    raise ValueError("pk cannot be negative values.")
```

Reply:
numpy.greater_equal might be an option as well

Reply:
All requests are reflected.

Reply:
@akahard2dj should there be a unit test of the `ValueError` now (correctly) thrown?

Reply:
Could you explain more detailed unit test of the `ValueError`?